### PR TITLE
Assignment 8 WIP

### PR DESCRIPTION
**Pull request by Ashrom0**

### CODE REVIEW

**Comment by TechGlue** [comment]
Will review :)

File: SecretSanta/test/SecretSanta.Business.Tests/GroupRepositoryTests.cs [hunk 1]
```diff
@@ -132,13 +132,13 @@ public void GenerateAssignments_WithLessThanThreeUsers_ReturnsError()
                 Name = "Group"
             });
 
-            AssignmentResult result = sut.GenerateAssignments(42);
+            AssignmentResult result = sut.Assignment(42);
 
-            Assert.AreEqual($"Group Group must have at least three users", result.ErrorMessage);
+            Assert.AreEqual("Group Group must have at least three users", result.ErrorMessage);
```
Comment (TechGlue) [inline SecretSanta/test/SecretSanta.Business.Tests/GroupRepositoryTests.cs:137]: There is a small typo here that makes the tests fail. Remove the extra Group in the string and you should be good.

**Review by TechGlue — COMMENTED** [review]
Assignment
* Update the GroupRepository with functionality to generate gift assignments. ✔
    * A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
    * A user is not allowed to be both the Giver and Recipient of the assignment. ✔
* Update the GroupsController with an endpoint for generating the assignments. ✔
* Update the Groups/Edit page to contain the UI elements for the following
    * A means to generate assignments. ✔
    * A means to display any errors that occurs while generating assignments. ✔
    * Display of the assignments for each of the users. ✔
* Write appropriate unit tests for changes made to the GroupRepository ✔
* Write appropriate functional tests for the change made to the GroupsController ✔

Minor error, but other than that Everything looks good. Great Work!

File: SecretSanta/test/SecretSanta.Web.EndToEnd/EndToEndTests.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
+using System.Threading.Tasks;
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Web.EndToEnd/EndToEndTests.cs:1]: Did you mean to include this file? It does not appear to be related to the assignment.

**Review by Keboo — COMMENTED** [review]
# Assignment
- Update the `GroupRepository` with functionality to generate gift assignments. ✔
  - A group with with 2 or fewer users should result in an error. This error should be displayed to a user. ✔
  - A user is not allowed to be both the Giver and Recipient of the assignment. ✔
- Update the GroupsController with an endpoint for generating the assignments. ✔
- Update the Groups/Edit page to contain the UI elements for the following
  - A means to generate assignments. ✔
  - A means to display any errors that occurs while generating assignments. ✔
  - Display of the assignments for each of the users. ✔
- Write appropriate unit tests for changes made to the `GroupRepository` ❌ - This is mostly done, however a test is failing due to not having the error message update.
- Write appropriate functional tests for the change made to the `GroupsController` ✔
